docs/help/FAQ/Puppet.rst

```diff
@@ -15,6 +19,40 @@ This will affect runs of ``puppet agent -t --debug`` as well as ``facter -p``.
 It is highly likely that you will have one of these values from the ``shmall``
 fact provided by the ``simplib`` module.
 
+.. _Bignum: https://ruby-doc.org/core-2.2.0/Bignum.html
 .. _FACT-1732: https://tickets.puppetlabs.com/browse/FACT-1732
 .. _Facter 3: https://docs.puppet.com/facter/3.8/
-.. _Bignum: https://ruby-doc.org/core-2.2.0/Bignum.html
+
+.. _faq-puppet-generate_types:
+
+When Should I Run `puppet generate types`?
+------------------------------------------
+
+The `puppet generate types`_ command was added to help solve the `SERVER-94`_
+:term:`puppet environment` isolation issue by caching :term:`custom type`
+definitions in each environment.
+
+SIMP has a Puppet class called ``pupmod::master::generate_types``, which is
+enabled by default, and uses ``incron`` to automatically run ``puppet generate
+types`` in the following cases:
+
+  * The ``puppet`` or ``puppetserver`` binaries have been updated.
+  * A new environment is added to the system.
+
+You will need to run ``puppet generate types`` manually if the following occurs:
+
+  * A new module is added to your environment that includes custom types.
+  * An existing custom type has its code modified.
+
+.. NOTE::
+   Previous versions of SIMP tried to include all of the cases above to ensure
```

Comment:
I would list the versions of SIMP and/or pupmod-simp-pupmod that did the old behavior, because, as soon as this documentation change is published, users will think this (yet to be released capability) is current.